octree/include/pcl/octree/octree_iterator.h

```diff
@@ -524,6 +523,77 @@ namespace pcl
       };
 
     //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    /** \brief @b Octree iterator class
+     * \note This class implements a forward iterator for traversing octrees in a breadth-first manner.
+     *       It only returns the node at a given depth in the octree, if any.
+     * \ingroup octree
+     * \author Fabien Rozar (fabien.rozar@gmail.com)
+     */
+    template<typename OctreeT>
+      class OctreeBreadthFirstAtDepthIterator : public OctreeBreadthFirstIterator<OctreeT>
+      {
+      public:
+
+        // public typedefs
+        typedef typename OctreeBreadthFirstIterator<OctreeT>::BranchNode BranchNode;
+        typedef typename OctreeBreadthFirstIterator<OctreeT>::LeafNode LeafNode;
```

Comment:
It's a matter of style but you can switch the typedefs to `using`. It's semantically more correct.